device/sycl/src/seeding/seed_finding.sycl

```diff
@@ -55,6 +60,39 @@ class select_seeds;
 
 }  // namespace kernels
 
+template <class kernelName>
+vecmem::data::vector_buffer<device::prefix_sum_element_t> make_prefix_sum_buff(
+    const std::vector<device::prefix_sum_size_t>& sizes, vecmem::copy& copy,
+    const traccc::memory_resource& mr, ::sycl::queue& queue) {
```

Comment:
The function should not be implemented like this, as a template. You shouldn't give different "names" for the different prefix filling kernels.

Just like how with CUDA we can have a single `traccc::cuda::kernels::fill_prefix_sum` global function, you should define a single `traccc::sycl::kernels::fill_prefix_sum` class in this file as well, and then use that to identify every kernel submission from this function. Since the kernel is really the same in all cases, the same code just operates on different inputs. (Note that this has a real performance impact. Since the SYCL runtime separately compiles and loads kernels with different names/identifiers at runtime. So making them share the same identifier will have benefits for the code.)